src/util/psbt/map/global.rs

```diff
@@ -0,0 +1,172 @@
+// Rust Bitcoin Library
+// Written by
+//   The Rust Bitcoin developers
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+use std::collections::HashMap;
+use std::io::Cursor;
+
+use blockdata::transaction::Transaction;
+use consensus::encode::{self, Decodable, Decoder};
+use util::psbt::map::Map;
+use util::psbt::raw;
+use util::psbt::serialize::Serialize;
+use util::psbt;
+use util::psbt::Error;
+
+/// A key-value map for global data.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Global {
+    /// The unsigned transaction, scriptSigs and witnesses for each input must be
+    /// empty.
+    pub unsigned_tx: Transaction,
+    /// Unknown global key-value pairs.
+    pub unknown: HashMap<raw::Key, Vec<u8>>,
+}
+
```

Comment:
Because abstractly, I think psbt errors are a subset of encode errors, and those functions sometimes will encounter encode errors that aren't psbt errors.

Reply:
Well, if a function always returns the smallest subset of errors it can produce, it should be able to know, no?

Probably all of the methods that deal with serialization with have to be able to handle `encode::Error`. But personally I still favor smallest subset in general. If consistency is favored instead, I guess that's ok.

Reply:
Which functions can you change to `psbt::Error`?

Reply:
Hmm, seemingly only from the two `from_unsigned_tx` methods.

Reply:
Fixed, thanks!